src/responder/kcm/kcm.c

```diff
@@ -311,21 +312,77 @@ static int kcm_process_init(TALLOC_CTX *mem_ctx,
     return ret;
 }
 
+static errno_t load_configuration(const char *config_file,
+                                  const char *config_dir,
+                                  const char *only_section)
+{
+    errno_t ret;
+    TALLOC_CTX *tmp_ctx;
+    uid_t sssd_uid;
+    gid_t sssd_gid;
+    struct confdb_ctx *cdb;
+    char *cdb_file;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_FATAL_FAILURE, "Failed to allocate the initial context\n");
+        return ENOMEM;
+    }
+
+    cdb_file = talloc_asprintf(tmp_ctx, "%s/%s", DB_PATH, CONFDB_KCM_FILE);
+    if (cdb_file == NULL) {
+        DEBUG(SSSDBG_FATAL_FAILURE, "Failed to allocate memory for the filename\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = confdb_setup(tmp_ctx, cdb_file, config_file, config_dir, only_section,
+                       &cdb);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_FATAL_FAILURE, "Unable to setup ConfDB [%d]: %s\n",
+             ret, sss_strerror(ret));
+        goto done;
+    }
+
+    /* Allow configuration database to be accessible
+     * when SSSD runs as nonroot */
+    sss_sssd_user_uid_and_gid(&sssd_uid, &sssd_gid);
```

Comment:
At the moment we don't even pretend `sssd_kcm` could be run under non-root.

And I'm getting rid of similar section in 'monitor' for non-root case anyway.

Let's skip it.

Reply:
Anyway, 'monitor' does this to allow other services to read 'config.ldb'
In case of `sssd_kcm`, it prepares file only for own consumption.

Reply:
OK. I will remove it. I just tried to keep the exact same behavior.